vite.config.ts

```diff
@@ -1,7 +1,40 @@
-import { defineConfig } from 'vite'
-import react from '@vitejs/plugin-react'
+/* eslint-disable import/no-extraneous-dependencies */
+
+/// <reference types="vitest" />
+/// <reference types="vite/client" />
+import checker from 'vite-plugin-checker';
+import { defineConfig } from 'vite';
+import react from '@vitejs/plugin-react';
+import stylelint from 'vite-plugin-stylelint';
 
 // https://vitejs.dev/config/
 export default defineConfig({
-  plugins: [react()],
-})
+  plugins: [
+    react(),
+    checker({
+      typescript: true,
+    }),
+    stylelint({
+      fix: true,
+      lintOnStart: true,
+    }),
+  ],
+  test: {
+    globals: true,
+    watch: true,
+    environment: 'jsdom',
+    setupFiles: './src/test/setup.ts',
+  },
+  resolve: {
+    alias: {
+      assets: '/src/assets',
```

Comment:
hm, I've never seen an approach like this. The typical approach use '@': '/src'. because importing a router from 'router' looks like importing from node modules for example.